Objects/setobject.c

```diff
@@ -757,6 +757,18 @@ frozenset_hash(PyObject *self)
     return hash;
 }
 
+static Py_hash_t
+set_hash(PyObject *self)
+{
+    PySetObject *so = (PySetObject *)self;
+    Py_uhash_t hash;
+    Py_BEGIN_CRITICAL_SECTION(self);
+    hash = frozenset_hash(self);
+    so->hash = (Py_hash_t)-1;
```

Comment:
Instead of this, why not refactor `frozenset_hash` and extract the code that is only responsible for computing the hash (without setting so->hash)

Reply:
OK, will do it.